commit c56f8ab6ca6ca0bdb9536fcce1ef24f1ef40ddc7
Author: Benoit Foucher <benoit@zeroc.com>
Date:   Mon Jun 27 17:54:30 2016 +0200

    Refactored SSL and iAP transports, support for running SSL on top
    of TCP/iAP/Bluetooth.

diff --git a/cpp/src/slice2cpp/Gen.cpp b/cpp/src/slice2cpp/Gen.cpp
index 81c335ad2d..5fe808581f 100644
--- a/cpp/src/slice2cpp/Gen.cpp
+++ b/cpp/src/slice2cpp/Gen.cpp
@@ -2527,7 +2527,7 @@ Slice::Gen::ObjectVisitor::visitClassDefStart(const ClassDefPtr& p)
     }
 
     bool hasBaseClass = !bases.empty() && !bases.front()->isInterface();
-    bool override = p->canBeCyclic() && (!hasBaseClass || !bases.front()->canBeCyclic());
+    bool override = !p->isLocal() && p->canBeCyclic() && (!hasBaseClass || !bases.front()->canBeCyclic());
     bool hasGCObjectBaseClass = basePreserved || override || preserved;
     if(!basePreserved && (override || preserved))
     {
